### DISCUSSION TITLE
How to apply styling?

### DISCUSSION THREAD

**Discussion by chiranjib-datta** [react-querybuilder/react-querybuilder#203] [Q&A] [answered]
How can i apply theme?

**Comment by jakeboone02** [comment, marked as answer]
There are no "themes" as such for this library, however it does use CSS class names that can be leveraged to style the components.  For example, the outer-most `div` has a class of `"queryBuilder"`, and each group `div` has a class of `"ruleGroup"`.  The fields lists, operators, values, and other components all have their own classes.

Additionally, the [demo](https://react-querybuilder.js.org/react-querybuilder/) [code](https://github.com/react-querybuilder/react-querybuilder/tree/main/packages/demo) has examples of different styling libraries being used in conjunction with this library.  Please note that with some of them you have to use custom components (see [here](https://github.com/react-querybuilder/react-querybuilder/tree/main/packages/demo/src/components)).

### UPDATE 2022:

React Query Builder now has officially supported component packages compatible with several popular style libraries:

- [Ant Design](https://ant.design/) ([@react-querybuilder/antd](https://www.npmjs.com/package/@react-querybuilder/antd))
- [Bootstrap](https://getbootstrap.com/) ([@react-querybuilder/bootstrap](https://www.npmjs.com/package/@react-querybuilder/bootstrap))
- [Bulma](https://bulma.io/) ([@react-querybuilder/bulma](https://www.npmjs.com/package/@react-querybuilder/bulma))
- [Chakra UI](https://chakra-ui.com/) ([@react-querybuilder/chakra](https://www.npmjs.com/package/@react-querybuilder/chakra))
- [MUI](https://mui.com/) ([@react-querybuilder/material](https://www.npmjs.com/package/@react-querybuilder/material))
